tools/payments/report_test.go

```diff
@@ -146,3 +142,127 @@ func TestPreparedReport_EnsureUniqueDest(t *testing.T) {
 		})
 	}
 }
+
+func TestAttestedReport_EnsureTransactionAmountsMatch(t *testing.T) {
+	type attestedTestCase struct {
+		name  string
+		given payments.AttestedReport
+		exp   error
+	}
+	type preparedTestCase struct {
+		given payments.PreparedReport
+	}
```

Comment:
Nitpick: Are not these types excessively verbose? We have already established the context by the test function name. What uncertainty do the extra prefixes on the types address? 😉